swarm/CHANGELOG.md

```diff
@@ -90,6 +90,8 @@
 - Deprecate methods `Swarm::with_executor`, `Swarm::with_*_executor`, `Swarm::without_executor`.
   Introduce similar methods in `SwarmBuilder`. See [PR 3588].
 
+- Gracefully disable oneshot handler on dial upgrade errors. See [PR 3577].
```

Comment:
Under the assumption that this is not a breaking change, this pull request should have bumped `libp2p-swarm` to `v0.42.1` and this changelog entry should have been under the `v0.42.1` section.

Reply:
We can just fix it in a new PR.